lib/senders/email.js

```diff
@@ -841,8 +845,12 @@ module.exports = function (log, config) {
   Mailer.prototype.postChangePrimaryEmail = function (message) {
     log.trace({ op: 'mailer.postChangePrimaryEmail', email: message.email, uid: message.uid })
 
+    const query = {}
+    if (message.email) { query.email = message.email }
+    if (message.uid) { query.uid = message.uid }
+
     var templateName = 'postChangePrimaryEmail'
-    var links = this._generateLinks(this.accountSettingsUrl, message.email, {}, templateName)
+    var links = this._generateLinks(this.accountSettingsUrl, message.email, query, templateName)
```

Comment:
The duplication of account link generation in so many areas seems ripe for extraction to save us from our future selves. It doesn't seem like `query` is used anywhere else, can the majority of the duplicate code be extracted to something like:

```js
var templateName = 'postChangePrimaryEmail';
var links = this._generateSettingsUrl(message, templateName);
```